core/extraction/towns_fund.py

```diff
@@ -65,9 +68,54 @@ def ingest_towns_fund_data(df_ingest: pd.DataFrame) -> Tuple[Dict[str, pd.DataFr
         project_lookup,
         programme_id,
     )
-    reporting_period = df_ingest["1 - Start Here"].iloc[4, 1]
 
-    return towns_fund_extracted, reporting_period
+    return towns_fund_extracted
+
+
+def extract_submission_details(submission_period: str) -> pd.DataFrame:
+    """
+    Create submission information and return in a DataFrame
+
+    Create submission info from submission period string, parsed from ingest form.
+
+    :param submission_period: String representation of a datetime period.
+    :return: DataFrame containing submission detail data.
+    """
+    # Data (strings) hard-coded, copied directly from TF_Reporting_Template.
+    funding_round = dict()
+    first_period = "1 April 2019"
+    for period, reporting_round in {
+        "2019/20 to 31 March 2022": 1,
+        "1 April 2022 to 30 September 2022": 2,
+        "1 October 2022 to 31 March 2023": 3,
+        "1 April 2023 to 30 September 2023": 4,
+        "1 October 2023 to 31 March 2024": 5,
+        "1 April 2024 to 30 September 2024": 6,
+        "1 October 2024 to 31 March 2025": 7,
+        "1 April 2025 to 30 September 2025": 8,
+        "1 October 2025 to 31 March 2026": 9,
+    }.items():
+        start_str, end_str = period.split(" to ")
+
+        # assuming start date of 1st round is 1st April 2019, otherwise extract from string
+        start_date = (
+            datetime.strptime(start_str, "%d %B %Y")
+            if start_str != "2019/20"
+            else datetime.strptime(first_period, "%d %B %Y")
+        )
+        end_date = datetime.strptime(end_str, "%d %B %Y")
+
+        funding_round[period] = {
+            "Submission Date": datetime.now(),
+            "Reporting Period Start": start_date,
+            "Reporting Period End": end_date,
+            "Reporting Round": str(reporting_round),
+        }
+
+    # TODO: Add filename and pickled submission file here?
```

Comment:
This data will be added straight to the model object - doesn't need to be part of this df